core/src/main/scala/kafka/server/metadata/BrokerMetadataListener.scala

```diff
@@ -153,40 +143,13 @@ class BrokerMetadataListener(config: KafkaConfig,
     }
   }
 
+  override def handleCommits(lastOffset: Long, messages: util.List[ApiMessage]): Unit = {
+    put(MetadataLogEvent(messages, lastOffset))
+  }
+
   def put(event: BrokerMetadataEvent): QueuedEvent = {
-    val queuedEvent = new QueuedEvent(event, time.nanoseconds())
-    dequeLock.lock()
-    try {
-      event match {
-        case _: OutOfBandRegisterLocalBrokerEvent |
-             _: OutOfBandFenceLocalBrokerEvent =>
-          // we have to check to see if any out-of-band events are already at the front of the deque;
-          // if so, then we need to add this event after those
-          val listBuffer: ListBuffer[QueuedEvent] = new ListBuffer()
-          var nextEvent = deque.peekFirst()
-          while (nextEvent != null && {
-            nextEvent.event match {
-              case _: OutOfBandRegisterLocalBrokerEvent |
-                   _: OutOfBandFenceLocalBrokerEvent => true
-              case _ => false
-            }}) {
-            listBuffer += deque.pollFirst()
-            nextEvent = deque.peekFirst()
-          }
-          // we've removed any out-of-band events that were there
-          // so now add the one we want to add
-          deque.addFirst(queuedEvent)
-          // and then put back any out-of-band events that were there before (keeping their order unchanged)
-          listBuffer.reverseIterator.foreach(eventAlreadyThere => deque.addFirst(eventAlreadyThere))
-        case _ =>
-          // add this non-out-of-band message to the end of the deque
-          deque.addLast(queuedEvent)
-      }
-      dequeNonEmptyCondition.signal()
-      dequeSize += 1
-    } finally {
-      dequeLock.unlock()
-    }
+    val queuedEvent = new QueuedEvent(event, time.milliseconds())
```

Comment:
Oh yeah, that's why I added the `max` when reporting the value. The use of nanoseconds confused me a little bit because the reported value was in milliseconds.